src/ck-core/cklocation.h

```diff
@@ -72,6 +72,14 @@ typedef enum : uint8_t
 PUPbytes(CkDeliver_t)
 
     class CkArrayOptions;
+
+struct CkLocEntry {
+  CmiUInt8 id = 0;
```

Comment:
Yes and no. When using full 64bit IDs, yes, 0 is the "invalid" ID. However, with the way CkLocMgr and CkArray current deal with 64bit IDs, it is not, because they don't use full IDs, only the element bits. Which is odd. This will change in the upcoming 64bit ID update, but for now I think this is fine. CkLocMgr and CkArray don't rely on checking this value for validity, and any other users of CkLocCache would be using full IDs, so this should work as intended for them.